### PR TITLE
FormResolveConflicts: Improve help when merging/rebasing

### PR DESCRIPTION
**Pull request by pmiossec**
## Proposed changes

When conflicts during merge/rebase, it is not always easy to what is "local" or "remote" ( See [here](https://jvns.ca/blog/2023/11/01/confusing-git-terminology/#ours-and-theirs-while-merging-or-rebasing) and [here](https://nitaym.github.io/ourstheirs/) )

- Add also terms "current"/"incoming" (that I found a little more explicit) and that user could be used to see if merging with "vscode"
- Add more explicit help as tooltips

### Before

* Merge:

image

* Rebase:

image

### After

* Merge:

image

* Rebase:

image

## Test methodology 

- Manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 199b9935fa10cea107ffcff6d2d14c54d85cc2f4
- Git 2.44.0.windows.1
- Microsoft Windows NT 10.0.22631.0
- .NET 8.0.1
- DPI 96dpi (no scaling)
- Portable: False

## Merge strategy

I agree that the maintainer **rebase** merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by RussKie — APPROVED** [review]

**Comment by gerhardol** [comment]
I would prefer to have the branch name where available, then commit, then remote/incoming (keep ours/theirs as it is used by Git).
Adding incoming will make the text even longer.

**Comment by mstv** [comment]
👍 When rebasing, the term "incoming (ours)" is much clearer to me than just "remote (ours)".
"remote/" and "local/" help to identify the regarding files in the merge tool.

It will be most helpful to me to have these additions in the lower pane of the dialog (green arrows). This can be looked up without closing the merge tool.
The help text (yellow arrows) would not benefit very much. It even might become too long.

image

**Comment by pmiossec** [comment]
> It will be most helpful to me to have these additions in the lower pane of the dialog (green arrows). This can be looked up without closing the merge tool.

Done.

**Comment by pmiossec** [comment]
> I would prefer to have the branch name where available, then commit, then remote/incoming (keep ours/theirs as it is used by Git).

I keep that in mind for later as it is a good idea but it's not easy to retrieve these information as we have no context when the form is opened. So I will definitively not do it in this PR which goal was a quick improvement on complaints I see here and there...

**Comment by gerhardol** [comment]
> > I would prefer to have the branch name where available, then commit, then remote/incoming (keep ours/theirs as it is used by Git).
> 
> I keep that in mind for later as it is a good idea but it's not easy to retrieve these information as we have no context when the form is opened. So I will definitively not do it in this PR which goal was a quick improvement on complaints I see here and there...

I looked at this some years ago and saw no quick fix, I was hoping you had one...

I am OK with the change, even if I personally see no improvement. 
The GE specific local/remote denominations could be replaced with incoming/current, that are better names (it is just having three names for the same thing that I do not like).

File: GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs [hunk 1]
```diff
@@ -329,9 +330,9 @@ private void InitializeComponent()
             label7.Location = new Point(3, 0);
             label7.Name = "label7";
             label7.Padding = new Padding(0, 3, 0, 3);
-            label7.Size = new Size(37, 22);
+            label7.Size = new Size(80, 21);
             label7.TabIndex = 1;
-            label7.Text = "Local";
+            label7.Text = "Local/Current";
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs:335]: In dependency on the operation:

```suggestion
            label7.Text = "Local/current (ours)";
```

```suggestion
            label7.Text = "Local/current (theirs)";
```

File: GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs [hunk 2]
```diff
@@ -371,9 +372,9 @@ private void InitializeComponent()
             label5.Location = new Point(3, 44);
             label5.Name = "label5";
             label5.Padding = new Padding(0, 3, 0, 3);
-            label5.Size = new Size(52, 22);
+            label5.Size = new Size(104, 21);
             label5.TabIndex = 5;
-            label5.Text = "Remote";
+            label5.Text = "Remote/Incoming";
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs:377]: ditto
```suggestion
            label5.Text = "Remote/incoming (...)";
```

File: GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs (hunk 1 above)
Comment (pmiossec) [reply GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs]: Done. Required a refactoring (introducing class `DisplayWithSuffixUpdater`) to do the same thing we did to display keyboard shortcuts in tooltips.

Also:
* added a tooltip:
image

& 

image
* fixed documentation link that was broken
* 2 other opportune refactorings

File: GitUI/CommandsDialogs/FormResolveConflicts.cs [hunk 3]
```diff
@@ -703,27 +729,12 @@ private void Reset_Click(object sender, EventArgs e)
...
         private string GetShortHash(ConflictedFileData item)
-        {
-            if (item.ObjectId is null)
-            {
-                return "@" + _deleted.Text;
-            }
-
-            return '@' + item.ObjectId.ToShortString();
-        }
+            => item.ObjectId is null ? "@" + _deleted.Text : '@' + item.ObjectId.ToShortString();
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormResolveConflicts.cs:737]:
```suggestion
            => $"@{item.ObjectId is null ? _deleted.Text : item.ObjectId.ToShortString()}";
```

**Review by mstv — COMMENTED** [review]
Thank you, just nits

File: GitExtUtils/DisplayWithSuffixUpdater.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿namespace GitExtUtils;
+
+public static class DisplayWithSuffixUpdater
+{
+    // NO-BREAK SPACE
```
Comment (mstv) [inline GitExtUtils/DisplayWithSuffixUpdater.cs:5]:
```suggestion
```

Comment (pmiossec) [reply GitExtUtils/DisplayWithSuffixUpdater.cs:5]: oups!

**Review by mstv — APPROVED** [review]

**Pull request merged**